From 3cf7de7259ff84cc0a0452330d6f3dd489ea68b3 Mon Sep 17 00:00:00 2001
From: John Cox <jc@kynesim.co.uk>
Date: Fri, 10 Jun 2022 15:32:40 +0000
Subject: [PATCH 28/41] caca: Set driver to ncurses avoiding segfault

For some reason it unloads GL module on error without telling anything
else and this creates a later segfault. Forcing ncurses stops this.
---
 modules/video_output/caca.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/modules/video_output/caca.c
+++ b/modules/video_output/caca.c
@@ -160,7 +160,11 @@ static int Open(vlc_object_t *object)
     }
 
     const char *driver = NULL;
-#ifdef __APPLE__
+// RPI: If driver is NULL then if we have X but DISPLAY is unset then somehow
+// the GL module becomes unloaded without anything noticing and that then
+// causes a segfault.
+//#ifdef __APPLE__
+#if 1
     // Make sure we don't try to open a window.
     driver = "ncurses";
 #endif
